media_player.kodi extra attributes for tvshow and music media

mvillarejo opened the pull request:
## Description:
New media attributes for kodi media players. 
(First pull request btw)

**Related issue (if applicable):** fixes #6250 

**Pull request in [home-assistant.github.io](https://github.com/home-assistant/home-assistant.github.io) with documentation (if applicable):** home-assistant/home-assistant.github.io#<home-assistant.github.io PR number goes here>

## Example entry for `configuration.yaml` (if applicable):
```yaml
media_player:
  # https://home-assistant.io/components/media_player.kodi/
  - platform: kodi
    host: osmc-01
    port: 8080
    name: kodi_living_room
    timeout: 10
```

## Checklist:

If user exposed functionality or configuration variables are added/changed:
  - [ ] Documentation added/updated in [home-assistant.github.io](https://github.com/home-assistant/home-assistant.github.io)

If the code communicates with devices, web services, or third-party tools:
  - [ ] Local tests with `tox` run successfully. **Your PR cannot be merged unless tests pass**
  - [ ] New dependencies have been added to the `REQUIREMENTS` variable ([example][ex-requir]).
  - [ ] New dependencies are only imported inside functions that use them ([example][ex-import]).
  - [ ] New dependencies have been added to `requirements_all.txt` by running `script/gen_requirements_all.py`.
  - [ ] New files were added to `.coveragerc`.

If the code does not interact with devices:
  - [ ] Local tests with `tox` run successfully. **Your PR cannot be merged unless tests pass**
  - [ ] Tests have been added to verify that the new code works.

[ex-requir]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard.py#L14
[ex-import]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard.py#L54

Commit:
media_player.kodi extra attributes for tvshow and music media

homeassistant commented:
Hi @mvillarejo, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;home-assistant#6622).

Once you do that we will be able to review and accept this pull request.

Thanks!

mvillarejo committed:
removed extra whitespaces/CR

homeassistant commented:
Hi @mvillarejo, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;home-assistant#6622).

Once you do that we will be able to review and accept this pull request.

Thanks!

emlove reviewed (dismissed):
LGTM after linter fixes. I assume you've tested this with several different media types?

Nice and small PR is 👍

armills commented:
Inspecting this a little further, it looks like the type we're getting doesn't tell us if music or a tvshow is playing: http://kodi.wiki/view/JSON-RPC_API/v6#Player.Type

With this change, do you see the extra info on the frontend, or is it just in the attributes?

mvillarejo commented:
Hold on a sec, looking into that now

armills commented:
I suspect what we need to do is add `'type'` to the [GetItem](http://kodi.wiki/view/JSON-RPC_API/v6#List.Item.Base) list, and change [`media_content_type`](https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/media_player/kodi.py#L345) to `return self._item.get('type')`.

Commit:
Kodi - add extra attributes #6250 (removed music attributes)

armills commented:
You can keep the music data here, it's good to expose all the info that we have. Also, you can drop the `None` from get. It's the default so there's no reason to copy it everywhere. The existing code wasn't a great example.

mvillarejo commented:
The thing is that if I left the attributes, movies attributes do not work as it can not find 'artist' attribute. What would you suggest?

armills commented:
Ah, I see.

I think that we should first implement the change described here: https://github.com/home-assistant/home-assistant/pull/6622#issuecomment-286529807 That should correctly give us the type of media that is playing.

We can then check the content type [after updating self._item here](https://github.com/home-assistant/home-assistant/pull/6622/files#diff-da43b164308888e738804f67589da788R305). If it's music, we can request the additional information by doing something like the following:
```python
if self.media_content_type == MEDIA_TYPE_MUSIC:
    self._item.update(yield from self.server.Player.GetItem(
        player_id, ['album', ...]
    ))['item']
```

You'll need to import `MEDIA_TYPE_MUSIC` from `homeassistant.components.media_player`. We should do the same thing for the tv show attributes.

Commit:
Restored music attributes, this is ready for merge

mvillarejo commented:
@armills what a mistake I made! 'artist' was duplicated and caused all the errors. I belive this is ready to go. Thanks!

armills commented:
OK cool. The simpler logic is better if it works.

You'll still need to fix the linting. [You can follow the instructions from the linter output.](https://travis-ci.org/home-assistant/home-assistant/jobs/211141613#L216)

mvillarejo committed:
linting amended

emlove reviewed (approved):
Looks good! We appreciate the contribution! 🎉

emlove committed:
Fix Kodi artist support

armills commented:
So I gave this a try on my system, and noticed two things:

1) Kodi sends artist info down in an array. I tweaked it to read the first item from the array.
2) Kodi has both album artist and artist. I added them both.

Can you give this a try on your system and make sure it works?

emlove committed:
Copy-paste error

mvillarejo commented:
yep, all good: media_album_artist: Radiohead; media_artist: Radiohead

Thanks for your help!

emlove committed:
Fix for non-music artist lookup

Kodi returns an emtpy list on videos, so we need to be able to
handle that as well.

armills commented:
OK, had to push one more fix. Sorry to clutter up your branch! 😆

mvillarejo commented:
ok, no worries
